File: AdvisoryLawyer/AdvisoryLawyer.Data/Models/Booking.cs
using System;
using System.Collections.Generic;

#nullable disable

namespace AdvisoryLawyer.Data.Models
{
    public partial class Booking
    {
        public Booking()
        {
            Advisories = new HashSet<Advisory>();
            Slots = new HashSet<Slot>();
        }

        public int Id { get; set; }
        public int CustomerId { get; set; }
        public DateTime BookingDate { get; set; }
        public string PaymentMethod { get; set; }
        public int TotalPrice { get; set; }
        public DateTime? PayDate { get; set; }
        public int Status { get; set; }
        public int CustomerCaseId { get; set; }
        public int LawyerId { get; set; }

        public virtual Customer Customer { get; set; }
        public virtual CustomerCase CustomerCase { get; set; }
        public virtual Lawyer Lawyer { get; set; }
        public virtual ICollection<Advisory> Advisories { get; set; }
        public virtual ICollection<Slot> Slots { get; set; }
    }
}

File: lib/lawyer/model_lawyer/booking.dart
class BookingDTO{
  int id;
  int customerId;
  String customerName;
  int lawyerId;
  String lawyerName;
  String bookingDate;
  String paymentMethod;
  int totalPrice;
  String payDate;
  int status;
  int customerCaseId;

  BookingDTO(
      {required this.id,
      required this.customerId,
      required this.customerName,
      required this.lawyerId,
      required this.lawyerName,
      required this.bookingDate,
      required this.paymentMethod,
      required this.totalPrice,
      required this.payDate,
      required this.status,
      required this.customerCaseId});

  factory BookingDTO.fromJson(Map<String, dynamic> json) {
    return BookingDTO(
      id : json['id'],
      customerId : json['customer_id'],
      customerName :json['customer_name'],
      lawyerId : json['lawyer_id'],
      lawyerName : json['lawyer_name'],
      bookingDate : json['booking_date'],
      paymentMethod : json['payment_method'],
      totalPrice : json['total_price'],
      payDate : json['pay_date'],
      status : json['status'],
      customerCaseId : json['customer_case_id']
    );
    
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
      data['id'] = this.id;
      data['customer_id'] = this.customerId;
      data['customer_name'] = this.customerName;
      data['lawyer_id'] = this.lawyerId;
      data['lawyer_name'] = this.lawyerName;
      data['booking_date'] = this.bookingDate;
      data['payment_method'] = this.paymentMethod;
      data['total_price'] = this.totalPrice;
      data['pay_date'] = this.payDate;
      data['status'] = this.status;
      data['customer_case_id'] = this.customerCaseId;
    return data;
  }



}